web/impact/impact/tests/test_algolia_api_key_view.py

```diff
@@ -53,10 +81,24 @@ def test_unauthenticated_user_is_denied(self):
                 response_data['detail'] == 'Authentication credentials '
                                            'were not provided.')
 
-    def test_staff_user_gets_empty_filters(self):
+    def test_entrepreneur_that_never_had_a_finalist_role_gets_403(self):
+        user = self._create_entrepreneur()
+        with self.settings(
+                ALGOLIA_APPLICATION_ID='test',
+                ALGOLIA_API_KEY='test'):
+            with self.login(email=user.email):
+                response = self.client.get(self.url)
+                self.assertTrue(response.status_code, 403)
+
+    def test_user_with_staff_role_grant_sees_all_mentors(self):
         user = self.basic_user()
-        user.is_staff = True
-        user.save()
+
+        named_group = NamedGroupFactory()
```

Comment:
I see this (96-100) repeated a few times, which suggests that it should be a function.

Reply:
Sorted this one, by moving to a function i.e. `_create_batch_program_and_named_group`